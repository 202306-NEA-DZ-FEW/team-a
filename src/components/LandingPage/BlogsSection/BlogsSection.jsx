import { motion } from "framer-motion";
import Link from "next/link";
import { useTranslation } from "next-i18next";

import BlogCard from "@/components/BlogCard";
import HorizontalCard from "@/components/BlogCard/HorizontalCard";
import Container from "@/components/container";

function BlogsSection({ blogs }) {
  const { t } = useTranslation();
  const container = {
    hidden: { opacity: 1, scale: 0 },
    visible: {
      opacity: 1,
      scale: 1,
      transition: {
        delayChildren: 0.3,
        staggerChildren: 0.2,
      },
    },
  };

  const item = {
    hidden: { y: 20, opacity: 0 },
    visible: {
      y: 0,
      opacity: 1,
    },
  };

  return (
    <Container className='lg:min-h-screen my-20 lg:my-0 flex flex-col justify-center'>
      <div></div>
      <motion.div
        className='flex flex-col lg:flex-row items-center justify-between mb-12 lg:mb-16 gap-y-8'
        initial={{ y: 30, opacity: 0 }}
        whileInView={{ y: 0, opacity: 1 }}
        transition={{ duration: 0.7, ease: "easeInOut" }}
      >
        <h2 className='text-3xl md:text-5xl font-bold'>Our Latest Blogs</h2>
        <Link
          className='btn btn-secondary rounded-full btn-md w-[45%] lg:w-[15%] text-current bg-opacity-50'
          href='/blogs'
        >
          Read More
        </Link>
      </motion.div>

      <motion.div
        className='grid grid-cols-3 gap-6 lg:gap-8 '
        variants={container}
        initial='hidden'
        whileInView='visible'
      >
        <motion.div
          className='col-span-3'
          variants={item}
          transition={{ duration: 0.7, ease: "easeInOut" }}
        >
          {blogs.length && <HorizontalCard {...blogs[0]} />}
        </motion.div>
        <motion.div
          className='col-span-3 flex flex-col lg:flex-row gap-x-8'
          variants={item}
          transition={{ duration: 0.7, ease: "easeInOut" }}
        >
          {blogs.slice(1, 3).map((blog) => (
            <BlogCard key={blog.id} {...blog} />
          ))}
          {!blogs.length && <p>{t("common:buttons:noItemsfound")}</p>}
        </motion.div>
      </motion.div>
    </Container>
  );
}

export default BlogsSection;
