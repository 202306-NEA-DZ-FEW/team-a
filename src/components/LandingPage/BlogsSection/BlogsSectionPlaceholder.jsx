function BlogsSectionPlaceholder() {
  return (
    <div className='flex max-w-6xl flex-col flex-wrap gap-8 justify-center items-center mx-auto w-full min-h-screen p-8'>
      <div className='flex justify-between w-full'>
        <div className='w-[300px] h-[35px] rounded-full bg-gray-300 animate-pulse mb-8'></div>
        <div className='w-[120px] h-[40px] rounded-full bg-gray-300 animate-pulse mb-8'></div>
      </div>
      <div className='w-full flex flex-wrap justify-center items-center gap-6 text-center'>
        <div className='flex-1 min-w-full flex-grow-0 h-80 bg-gray-300 animate-pulse rounded-2xl'></div>

        {[2, 3].map((i) => (
          <div
            key={i}
            className='flex-1 min-w-[300px] h-80 bg-gray-300 animate-pulse rounded-2xl'
          ></div>
        ))}
      </div>
    </div>
  );
}

export default BlogsSectionPlaceholder;
