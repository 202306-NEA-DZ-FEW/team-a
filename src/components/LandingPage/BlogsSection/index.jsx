import { motion } from "framer-motion";
import Link from "next/link";
import { useTranslation } from "next-i18next";
import { IoIosArrowDropleft, IoIosArrowDropright } from "react-icons/io";

import BlogCard from "@/components/BlogCard";
import HorizontalCard from "@/components/BlogCard/HorizontalCard";
import Container from "@/components/container";

function BlogsSection({ blogs }) {
  const { i18n, t } = useTranslation();

  return (
    <Container className='lg:min-h-screen my-20 lg:my-0 flex flex-col justify-center'>
      {/* Title & Button */}
      <motion.div
        className='flex flex-col lg:flex-row items-center justify-between mb-12 lg:mb-16 gap-y-8'
        initial={{ y: 30, opacity: 0 }}
        whileInView={{ y: 0, opacity: 1 }}
        transition={{ duration: 0.4 }}
        viewport={{ once: true }}
      >
        <h2 className='text-3xl md:text-5xl font-bold'>
          {t("common:latestBlogs")}
        </h2>
        <Link
          className='btn btn-secondary rounded-full btn-md min-w-[150px] lg:w-[15%] text-current bg-opacity-50 flex items-center gap-2'
          href='/blogs'
        >
          {t("common:buttons:readMore")}
          {i18n.language !== "ar" ? (
            <IoIosArrowDropright className='text-xl' />
          ) : (
            <IoIosArrowDropleft className='text-xl' />
          )}
        </Link>
      </motion.div>
      {/* Cards Section */}
      <div className='grid grid-cols-3 gap-6 lg:gap-8'>
        <motion.div
          className='col-span-3'
          initial={{ y: 30, opacity: 0, scale: 0.7 }}
          whileInView={{ y: 0, opacity: 1, scale: 1 }}
          transition={{ duration: 0.4, delay: 0.2 }}
          viewport={{ once: true }}
        >
          {blogs.length && <HorizontalCard {...blogs[0]} />}
        </motion.div>
        <motion.div
          className='col-span-3 flex flex-col lg:flex-row gap-y-6 gap-x-8'
          initial={{ y: 30, opacity: 0, scale: 0.7 }}
          whileInView={{ y: 0, opacity: 1, scale: 1 }}
          transition={{ duration: 0.4, delay: 0.6 }}
          viewport={{ once: true }}
        >
          {blogs.slice(1, 3).map((blog) => (
            <BlogCard key={blog.id} {...blog} />
          ))}
          {!blogs.length && <p>{t("common:buttons:noItemsFound")}</p>}
        </motion.div>
      </div>
    </Container>
  );
}

export default BlogsSection;
