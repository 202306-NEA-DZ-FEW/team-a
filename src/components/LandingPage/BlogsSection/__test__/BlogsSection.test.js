import renderer from "react-test-renderer";

import BlogsSection from "../BlogsSection";

it("renders correctly", () => {
  const mockBlogs = [
    {
      id: 1,
      title: "Hello",
      imageUrl: "https://placehold.co/600x400",
      createdAt: {},
      description: "Hello",
    },
  ];
  const tree = renderer.create(<BlogsSection blogs={mockBlogs} />).toJSON();
  expect(tree).toMatchSnapshot();
});
