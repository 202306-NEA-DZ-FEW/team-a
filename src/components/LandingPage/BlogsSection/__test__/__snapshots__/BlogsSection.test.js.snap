// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="max-w-6xl mx-auto px-8 xl:px-0 lg:min-h-screen my-20 lg:my-0 flex flex-col justify-center"
>
  <div />
  <div
    className="flex flex-col lg:flex-row items-center justify-between mb-12 lg:mb-16 gap-y-8"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(30px) translateZ(0)",
      }
    }
  >
    <h2
      className="text-3xl md:text-5xl font-bold"
    >
      Our Latest Blogs
    </h2>
    <a
      className="btn btn-secondary rounded-full btn-md w-[45%] lg:w-[15%] text-current bg-opacity-50"
      href="/blogs"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Read More
    </a>
  </div>
  <div
    className="grid grid-cols-3 gap-6 lg:gap-8 "
    style={
      Object {
        "opacity": 1,
        "transform": "scale(0) translateZ(0)",
      }
    }
  >
    <div
      className="col-span-3"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(20px) translateZ(0)",
        }
      }
    >
      <div
        className="card lg:card-side bg-base-100 shadow-xl h-full"
      >
        <figure
          className=" lg:w-[45%]"
        >
          <img
            alt="Blog Image"
            className="w-full"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height={400}
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=828&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=828&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={400}
          />
        </figure>
        <div
          className="card-body lg:w-[35%] p-6 lg:p-10 gap-4"
        >
          <h2
            className="card-title text-4xl"
            title="Hello"
          >
            Hello
          </h2>
          <p
            className="text-lg font-light"
            title="Hello"
          >
            Hello
          </p>
          <div
            className="card-actions flex gap-2 justify-between"
          >
            <span>
              Invalid date
            </span>
            <a
              className="text-secondary/[.7] flex items-center gap-2"
              href="/blogs/1"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              buttons.readMore
              <svg
                className="text-xl"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M210.7 147.6c7.5-7.5 19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81.1-81.9-81.1c-7.6-7.4-7.6-19.6 0-27.2z"
                />
                <path
                  d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm32 0c0-47 18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-span-3 flex flex-col lg:flex-row gap-x-8"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(20px) translateZ(0)",
        }
      }
    />
  </div>
</div>
`;
